### ISSUE TITLE
Flaky test velox_tool_trace_test FilterProjectReplayerTest.filterProject

### ISSUE THREAD

**Issue by jinchengchenghh** [facebookincubator/velox#11430]
### Bug description

Some PRs failed by this error.
https://github.com/facebookincubator/velox/actions/runs/11659126189/job/32459211067?pr=10797
https://github.com/facebookincubator/velox/actions/runs/11658215946/job/32456933493?pr=10711
```
[ RUN      ] FilterProjectReplayerTest.filterProject
E20241104 07:09:39.906908 64890 Exceptions.h:66] Line: /home/runner/work/velox/velox/velox/velox/common/file/FileInputStream.cpp:31, Function:FileInputStream, Expression: fileSize_ > 0 (0 vs. 0) Empty FileInputStream, Source: RUNTIME, ErrorCode: INVALID_STATE
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: (0 vs. 0) Empty FileInputStream
Retriable: False
Expression: fileSize_ > 0
Function: FileInputStream
File: /home/runner/work/velox/velox/velox/velox/common/file/FileInputStream.cpp
Line: 31
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxException5State4makeIZNS1_C4EPKcmS5_St17basic_string_viewIcSt11char_traitsIcEES9_S9_S9_bNS1_4TypeES9_EUlRT_E_EESt10shared_ptrIKS2_ESA_SB_
# 2  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 3  _ZN8facebook5velox17VeloxRuntimeErrorC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bS7_
# 4  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 5  _ZN8facebook5velox6common15FileInputStreamC2EOSt10unique_ptrINS0_8ReadFileESt14default_deleteIS4_EEmPNS0_6memory10MemoryPoolE
# 6  _ZSt11make_uniqueIN8facebook5velox6common15FileInputStreamEJSt10unique_ptrINS1_8ReadFileESt14default_deleteIS5_EEiRKPNS1_6memory10MemoryPoolEEENSt9_MakeUniqIT_E15__single_objectEDpOT0_
# 7  _ZNK8facebook5velox4exec5trace24OperatorTraceInputReader14getInputStreamEv
# 8  _ZN8facebook5velox4exec5trace24OperatorTraceInputReaderC1ENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt10shared_ptrIKNS0_7RowTypeEEPNS0_6memory10MemoryPoolE
# 9  _ZSt11make_uniqueIN8facebook5velox4exec5trace24OperatorTraceInputReaderEJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt10shared_ptrIKNS1_7RowTypeEEPNS1_6memory10MemoryPoolEEENSt9_MakeUniqIT_E15__single_objectEDpOT0_
# 10 _ZN8facebook5velox4exec5trace17OperatorTraceScanC2EiPNS1_9DriverCtxERKSt10shared_ptrIKNS0_4core13TraceScanNodeEE
# 11 _ZSt11make_uniqueIN8facebook5velox4exec5trace17OperatorTraceScanEJRmPNS2_9DriverCtxERKSt10shared_ptrIKNS1_4core13TraceScanNodeEEEENSt9_MakeUniqIT_E15__single_objectEDpOT0_
# 12 _ZN8facebook5velox4exec13DriverFactory12createDriverESt10unique_ptrINS1_9DriverCtxESt14default_deleteIS4_EESt10shared_ptrINS1_14ExchangeClientEESt8functionIFiiEE
# 13 _ZN8facebook5velox4exec4Task19createDriversLockedEj
# 14 _ZN8facebook5velox4exec4Task21createAndStartDriversEj
# 15 _ZN8facebook5velox4exec4Task5startEjj
# 16 _ZN8facebook5velox4exec4test23MultiThreadedTaskCursor5startEv
# 17 _ZN8facebook5velox4exec4test23MultiThreadedTaskCursor8moveNextEv
# 18 _ZN8facebook5velox4exec4test10readCursorERKNS2_16CursorParametersESt8functionIFvPNS1_4TaskEEEm
# 19 _ZN8facebook5velox4exec4test18AssertQueryBuilder10readCursorEv
# 20 _ZN8facebook5velox4exec4test18AssertQueryBuilder11copyResultsEPNS0_6memory10MemoryPoolERSt10shared_ptrINS1_4TaskEE
# 21 _ZN8facebook5velox4exec4test18AssertQueryBuilder11copyResultsEPNS0_6memory10MemoryPoolE
# 22 _ZN8facebook5velox4tool5trace20OperatorReplayerBase3runEv
# 23 _ZN8facebook5velox4tool5trace4test44FilterProjectReplayerTest_filterProject_Test8TestBodyEv
# 24 _ZN7testing8internal38HandleSehExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc
# 25 _ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc
# 26 _ZN7testing4Test3RunEv
# 27 _ZN7testing8TestInfo3RunEv
# 28 _ZN7testing9TestSuite3RunEv
# 29 _ZN7testing8internal12UnitTestImpl11RunAllTestsEv
# 30 _ZN7testing8internal38HandleSehExceptionsInMethodIfSupportedINS0_12UnitTestImplEbEET0_PT_MS4_FS3_vEPKc
# 31 _ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS0_12UnitTestImplEbEET0_PT_MS4_FS3_vEPKc
# 32 _ZN7testing8UnitTest3RunEv
# 33 _Z13RUN_ALL_TESTSv
# 34 main
# 35 0x0000000000029d8f
# 36 __libc_start_main
# 37 _start
" thrown in the test body.
```

### System information

CI

**Comment by duanmeng** [comment]
I will fix it in #11271

**Comment by jinchengchenghh** [comment]
Thanks very much!

**Comment by Yohahaha** [comment]
https://github.com/facebookincubator/velox/actions/runs/11684503115/job/32535884530?pr=11369

FilterProjectReplayerTest.projectOnly

**Comment by Yohahaha** [comment]
> I will fix it in #11271

@duanmeng may I ask does #11271 fix above issue?

**Comment by duanmeng** [comment]
> > I will fix it in #11271
> 
> @duanmeng may I ask does #11271 fix above issue?

@Yohahaha FYI, I've fixed it in a separate PR https://github.com/facebookincubator/velox/pull/11436.
